File: android/src/main/kotlin/voip24h/sdk/mobile/voip24h_sdk_mobile/model/SipConfiguration.kt
package voip24h.sdk.mobile.voip24h_sdk_mobile.model

import org.linphone.core.*

data class SipConfiguration(
    val extension: String,
    val domain: String,
    val password: String,
    var port: Int = 5060,
    var transportType: Int = TransportType.Udp.toInt(),
    var isKeepAlive: Boolean = false
) {
    fun toLpTransportType() : TransportType = TransportType.fromInt(transportType)
}

File: ios/Classes/model/SipConfiguration.swift
//
//  SipConfiguration.swift
//  voip24h_sdk_mobile
//
//  Created by Phát Nguyễn on 12/08/2022.
//

import Foundation
import linphonesw


class SipConfiguaration: Decodable {
    
    var ext: String!
    var password: String!
    var domain: String!
    var port: Int = 5060
    var transportType: Int = 0
    var isKeepAlive: Bool = false
    
    
    func toLpTransportType() -> TransportType {
        switch(transportType) {
            case 1:
                return TransportType.Tcp
            case 0:
                return TransportType.Udp
            case 2:
                return TransportType.Tls
            default:
                return TransportType.Udp
        }
    }
    
    private enum CodingKeys : String, CodingKey {
        case ext = "extension", password, domain, port, transportType, isKeepAlive
    }
}
